src/media/js/elements/select.js

```diff
@@ -0,0 +1,263 @@
+                        var option = options[i];
+
+                        var mktOption = document.createElement('mkt-option');
+                        copyAttrs(mktOption, option);
+                        mktOption.innerHTML = option.innerHTML;
+                        mktOption.option = option;
+                        mktOptGroup.appendChild(mktOption);
+                    }
+
+                    // Add the <optgroup>s.
+                    root.appendChild(mktOptGroup);
+                    mktOptGroup.optGroup = optGroup;
+                    select.appendChild(optGroup);
+
+                    // Add the hidden <select>
+                    root.appendChild(select);
+                    root.select = select;
+
+                    // Update indices (for colors).
+                    root.updateOptionIndices();
+
+                    // Align options with <mkt-selected-text>.
+                    if (root.alignOptions() > 0) {
+                        root.setAttribute('data-mkt-select--aligned', true);
+                    }
+
+                    // Set up click handlers.
+                    root.addEventListener('click', function(e) {
+                        root.toggle();
+                        if (e.target.tagName == 'MKT-OPTION') {
+                            // Change value if click on option.
+                            root.value = e.target.value;
+                        }
+                    });
+                }
+            },
+            select: {
+                // Actual <select> element to proxy to, steal its interface.
+                // Value set in the createdCallback.
+                value: null,
+                writable: true
+            },
+            toggle: {
+                // Toggles visibility of the <mkt-option>s.
+                value: function(bool) {
+                    if (!this.getAttribute('data-mkt-select--aligned')) {
+                        if (this.alignOptions() > 0) {
+                            this.setAttribute('data-mkt-select--aligned', true);
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            value: {
+                get: function() {
+                    return this.select.value;
+                },
+                set: function(value) {
+                    // Change value of hidden select element.
+                    var root = this;
+                    var select = root.select;
+                    select.value = value;
+
+                    // Change [selected] of <option>s.
+                    var options = select.querySelectorAll('option');
+                    var selectedOption = select.querySelector(
+                        'option[value="' + value +'"]');
```

Comment:
I don't think there's any way of a user specifying a `value` but this seems like it could be an XSS. It probably isn't really an issue but since we're going to loop over all of the options on the next line the loop may as well remove `selected` if `options[i].value !== value` and add it if `options[i].value === value`.